_stbt/core.py

```diff
@@ -0,0 +1,2329 @@
+        image = cv2.imread(template_name, cv2.CV_LOAD_IMAGE_COLOR)
+        if image is None:
+            raise UITestError("Failed to load template file: %s" %
+                              template_name)
+        return _AnnotatedTemplate(image, template, template_name)
+
+
+def _crop(frame, region):
+    if not _image_region(frame).contains(region):
+        raise ValueError("'frame' doesn't contain 'region'")
+    return frame[region.y:region.bottom, region.x:region.right]
+
+
+def _image_region(image):
+    return Region(0, 0, image.shape[1], image.shape[0])
+
+
+class MotionResult(namedtuple('MotionResult', 'timestamp motion')):
+    """The result from `detect_motion`.
+
+    * `timestamp`: Video stream timestamp.
+    * `motion`: Boolean result.
+    """
+    pass
+
+
+class OcrMode(IntEnum):
+    """Options to control layout analysis and assume a certain form of image.
+
+    For a (brief) description of each option, see the `tesseract(1)
+    <http://tesseract-ocr.googlecode.com/svn/trunk/doc/tesseract.1.html#_options>`_
+    man page.
+    """
+    ORIENTATION_AND_SCRIPT_DETECTION_ONLY = 0
+    PAGE_SEGMENTATION_WITH_OSD = 1
+    PAGE_SEGMENTATION_WITHOUT_OSD_OR_OCR = 2
+    PAGE_SEGMENTATION_WITHOUT_OSD = 3
+    SINGLE_COLUMN_OF_TEXT_OF_VARIABLE_SIZES = 4
+    SINGLE_UNIFORM_BLOCK_OF_VERTICALLY_ALIGNED_TEXT = 5
+    SINGLE_UNIFORM_BLOCK_OF_TEXT = 6
+    SINGLE_LINE = 7
+    SINGLE_WORD = 8
+    SINGLE_WORD_IN_A_CIRCLE = 9
+    SINGLE_CHARACTER = 10
+
+    # For nicer formatting of `ocr` signature in generated API documentation:
+    def __repr__(self):
+        return str(self)
+
+
+class TextMatchResult(namedtuple(
+        "TextMatchResult", "timestamp match region frame text")):
+
+    """The result from `match_text`.
+
+    * ``timestamp``: Video stream timestamp.
+    * ``match``: Boolean result, the same as evaluating `TextMatchResult` as a
+      bool. That is, ``if result:`` will behave the same as
+      ``if result.match:``.
+    * ``region``: The `Region` (bounding box) of the text found, or ``None`` if
+      no text was found.
+    * ``frame``: The video frame that was searched, in OpenCV format.
+    * ``text``: The text (unicode string) that was searched for, as given to
+      `match_text`.
+    """
+    # pylint: disable=E1101
+    def __nonzero__(self):
+        return self.match
+
+    def __str__(self):
+        return (
+            "TextMatchResult(timestamp=%s, match=%s, region=%s, frame=%s, "
+            "text=%s)" % (
+                self.timestamp,
+                self.match,
+                self.region,
+                "%dx%dx%d" % (self.frame.shape[1], self.frame.shape[0],
+                              self.frame.shape[2]),
+                repr(self.text)))
+
+
+def new_device_under_test_from_config(
+        gst_source_pipeline=None, gst_sink_pipeline=None, control_uri=None,
+        save_video=False, restart_source=False,
+        transformation_pipeline='identity'):
```

Comment:
How about `run.save_video`, `global.restart_source`, and `global.transformation_pipeline`?